### DISCUSSION TITLE
Cancelling periodic jobs question

### DISCUSSION THREAD

**Discussion by sheldondz** [riverqueue/river#359] [Q&A] [answered]
Is there a way to cancel periodic jobs using the job.ID? the documentation points to

```
// Remove a periodic job using handle return by Client.PeriodicJobs.Add.
riverClient.PeriodicJobs().Remove(periodicJobHandle)
```
Will this work even after the client is restarted? or is this only in memory?

**Comment by sheldondz** [comment]
I have tested this when client is running and it works fine, the periodic job no longer runs when removed using the handle.

**Comment by sheldondz** [comment]
After client restart the job stops running, so i am assuming for periodic jobs do we need to reload all the jobs on client restart?

**Comment by brandur** [comment, marked as answer]
It sounds like you got there already, but yes:

* Individual jobs generated from a periodic job configuration of course get persisted to the database, but periodic job configuration _does not_. If you want that to survive across restarts, either define it in code, or you'll need to store your own configuration somewhere out of band.
* The periodic job handles that are used to remove a periodic job configuration after being added are _not_ stable across restarts. They're only valid as long as the process is running.

**Comment by sheldondz** [comment]
We have the configuration stored so will add the code to reload all periodic jobs on startup. Thanks!
